pkg/templateservicebroker/servicebroker/provision.go

```diff
@@ -42,8 +42,14 @@ func (b *Broker) ensureSecret(u user.Info, namespace string, brokerTemplateInsta
 		Data: map[string][]byte{},
 	}
 
+	// allow empty string values for non-generated parameters.
+	// empty string values will be ignored for generated parameter/
 	for k, v := range preq.Parameters {
-		secret.Data[k] = []byte(v)
+		for _, param := range template.Parameters {
+			if param.Name == k && ((v == "" && param.Generate == "") || v != "") {
```

Comment:
style: we generally use `len(s) == 0` to check for an empty string

Reply:
thank you, I'll fix